src/harness/testcases/WINNF_FT_S_SCS_testcase.py

```diff
@@ -64,3 +66,75 @@ def test_WINNF_FT_S_SCS_5(self):
     Checks that a CBSD registration with this configuration succeed.
     """
     self.doTestCipher('ECDHE-RSA-AES128-GCM-SHA256')
+
+  def generate_SCS_12_default_config(self, filename):
+    """Generates the WinnForum configuration for SCS.12"""
+    # Create the actual config for client cert/key path
+    
+    config = { 
+        'clientCert': self.getCertFilename("client_expired.cert"),
+        'clientKey': self.getCertFilename("client_expired.key")
+    }
+    writeConfig(filename, config)
+  
+  @configurable_testcase(generate_SCS_12_default_config)
+  def test_WINNF_FT_S_SCS_12(self,config_filename):
+    """Expired certificate presented during registration.
+
+    Checks that SAS UUT response with fatal alert message.
+    """
+    config = loadConfig(config_filename)
+    self.assertTlsHandshakeFailure(client_cert=config['clientCert'],
+                                   client_key=config['clientKey'])
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_SCS_13(self):
+    """ Disallowed TLS method attempted during registration.
+
+    Checks that SAS UUT response with fatal alert message.
+    """
+    self.assertTlsHandshakeFailure(ssl_method=SSL.TLSv1_1_METHOD)
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_SCS_14(self):
+    """Invalid ciphersuite presented during registration.
+
+    Checks that SAS UUT response with fatal alert message.
+    """
+    self.assertTlsHandshakeFailure(ciphers='ECDHE-RSA-AES256-GCM-SHA384')
+
+  def generate_SCS_15_default_config(self, filename):
+    """ Generates the WinnForum configuration for SCS.15 """
+    # Create the actual config for client cert/key path
+    
+    config = {
+        'clientCert': self.getCertFilename("client_inapplicable.cert"),
+        'clientKey': self.getCertFilename("client.key")
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_SCS_15_default_config)
+  def test_WINNF_FT_S_SCS_15(self,config_filename):
+    """Certificate with inapplicable fields presented during registration. 
+    
+    Checks that SAS UUT response with tls connection succedded and response should be 104.    
+    """
+    self.SasReset()
+    config = loadConfig(config_filename)
+    
+    # load device certs and device_a from config file
+    device_a_cert = config['clientCert']
+    device_a_key = config['clientKey']
+    device_a = json.load(open(os.path.join('testcases', 'testdata', 'device_a.json')))
+    
+    # Inject fccId and userId
+    self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_a['userId']})
+    self.assertTlsHandshakeFailure(client_cert=device_a_cert,client_key=device_a_key)
```

Comment:
from the spec:
```
2.  CBSD Test Harness attempts to establish a TLS handshake with the SAS UUT using a valid ciphersuite
     CHECK: The SAS UUT response must satisfy all of the following conditions:
        - A valid TLS session is established
3.  CBSD Test Harness sends a valid registration request to the SAS UUT
     CHECK: ...response=104 ...
```

you should use self.assertTlsHandshakeSucceed and not check for a failure.
Also to have the handshake check occuring at first, I will be better to do:

```
def test_WINNF_FT_S_SCS_15(self,config_filename):
     """Certificate with inapplicable fields presented during registration. 
     
     Checks that SAS UUT response with TLS connection succeeded and response should be 104.    
     """
     config = loadConfig(config_filename)
     
     # load the keys/certs and check that TLS handshake is valid
     device_a_cert = config['clientCert']
     device_a_key = config['clientKey']
     self.assertTlsHandshakeSucceed(self._sas_admin._base_url, ['AES128-GCM-SHA256'], device_a_cert, device_a_key)
     
     # Load device and inject fccId and userId
     device_a = json.load(open(os.path.join('testcases', 'testdata', 'device_a.json')))
     self.SasReset()
     self._sas_admin.InjectFccId({'fccId': device_a['fccId']})
     self._sas_admin.InjectUserId({'userId': device_a['userId']})
  
     # Send registration Request with certs(inapplicable fields) to SAS UUT 
     request = {'registrationRequest': [device_a]}
     response = self._sas.Registration(request, ssl_cert=device_a_cert, ssl_key=device_a_key)['registrationResponse'][0]
 
     # Check registration response
     self.assertEqual(response['response']['responseCode'],104) 
```

Reply:
Norbert, thanks for the valuable inputs. Fixed now.